# HG changeset patch
# User Rik <rik@octave.org>
# Date 1373566617 25200
#      Thu Jul 11 11:16:57 2013 -0700
# Node ID 9c971fa62a770329f4146492bee781a5abfdff5b
# Parent  2ee5109d591459c7676fce74e8547db56153539c
trace.m: Update docstring.  Use isvector() in checking inputs.

* scripts/linear-algebra/trace.m: Update docstring.  Use isvector() in checking
inputs.  Add %!tests for vector.

diff --git a/scripts/linear-algebra/trace.m b/scripts/linear-algebra/trace.m
--- a/scripts/linear-algebra/trace.m
+++ b/scripts/linear-algebra/trace.m
@@ -12,43 +12,49 @@
 ## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 ## General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
-## @deftypefn {Function File} {} trace (@var{A})
-## Compute the trace of @var{A}, @code{sum (diag (@var{A}))}.
+## @deftypefn {Hello World} {} trace (@var{A})
+## @deftypefnx {Function File} {} trace (@var{A})
+## Compute the trace of @var{A}, the sum of the elements along the main
+## diagonal.
+##
+## The implementation is straightforward: @code{sum (diag (@var{A}))}.
+## @seealso{eig}
 ## @end deftypefn
 
 ## Author: jwe
 
 function y = trace (A)
 
   if (nargin != 1)
     print_usage ();
   endif
 
   if (ndims (A) > 2)
     error ("trace: only valid on 2-D objects");
   elseif (isempty (A))
     y = 0;
-  elseif (any (size (A) == 1))
+  elseif (isvector (A))
     y = A(1);
   else
     y = sum (diag (A));
   endif
 
 endfunction
 
 
 %!assert (trace ([1, 2; 3, 4]), 5)
 %!assert (trace ([1, 2; 3, 4; 5, 6]), 5)
 %!assert (trace ([1, 3, 5; 2, 4, 6]), 5)
 %!assert (trace ([]), 0)
-%!assert (trace (randn (1,0)), 0)
+%!assert (trace (rand (1,0)), 0)
+%!assert (trace ([3:10]), 3)
 
 %!error trace ()
 %!error trace (1, 2)
 %!error <only valid on 2-D objects> trace (reshape (1:9,[1,3,3]))
 
